### PR TITLE
[Firestore] Consolidate Firestore and FirestoreSwift

### PR DESCRIPTION
**Pull request by ncooke3**
**Context**
- The contents of FST's Swift extension SDK has been moved into the main FST module, and the FST Swift extension SDK now re-exports the API that used to live in it.
- This change should be non-breaking.

---

**Tasks**
- [ ] New Swift code moved into base SDK namespace should build on Firebase's minimum deployment version for each applicable platform (use client_app?)
- [x] The affected extension podspecs should be versioned to pin to the releasing version.
- [ ] When this PR is merged, stage the affected podspecs and notify the Games team.

---

**Important**
- This PR contains an assumption in the source code that docs will be at hosted at https://firebase.google.com/docs/ios/swift-migration cc: @morganchen12

### CODE REVIEW

**Comment by ncooke3** [comment]
Ah, I need to make FirebaseShareSwift a binary target for the SPM build...

**Comment by ncooke3** [comment]
The Firestore QS workflow had failed with a weird [error](https://github.com/firebase/firebase-ios-sdk/actions/runs/6181088595/job/16785986167?pr=11806#step:10:1161):
```
	Failed to build module 'FirebaseFirestoreSwift'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)', while this compiler is 'Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)'). Please select a toolchain which matches the SDK.
```

It's confusing because it's saying the SDK was built with Swift version 5.7.**1** but can't be compiled by Swift version 5.7.**2**...

**Comment by paulb777** [comment]
This is similar to errors we were seeing when we first tried to build the zip with Xcode 14.1 and were packaging the wrong includes.

**Comment by ncooke3** [comment]
fa1f3b9 fixed the qs zip failure with the undefined c++ symbols

**Comment by ncooke3** [comment]
https://github.com/firebase/firebase-ios-sdk/pull/11806/commits/c33db0d1ffef2b504787538cf6b9a33ae00d2ae1 should fix the remaining `firestore/spm` failure.

**Review by andrewheard — APPROVED** [review]

**Review by paulb777 — APPROVED** [review]

File: Package.swift [hunk 1]
```diff
@@ -616,13 +616,10 @@ let package = Package(
...
-    .target(
+    .binaryTarget(
       name: "FirebaseSharedSwift",
-      path: "FirebaseSharedSwift/Sources",
-      exclude: [
-        "third_party/FirebaseDataEncoder/LICENSE",
-        "third_party/FirebaseDataEncoder/METADATA",
-      ]
+      url: "https://dl.google.com/firebase/ios/experimental/bin/sharedswift/10.16.0/FirebaseSharedSwift.zip",
+      checksum: "cfa32eb16ba8fafa7a443ab8f48e019794470bb358066d6c571df1531df02fe6"
```
Comment (paulb777) [inline Package.swift:622]: Is there a corresponding release process update for this?

Comment (ncooke3) [reply Package.swift:622]: Yes, there will need to be. I'm thinking we can extend the tooling used to handle the Firestore binary to also handle SharedSwift.

**Pull request merged**